### ISSUE TITLE
Drag and drop does not work after a fresh start of Zotero 

### ISSUE THREAD

**Issue by rikeijin** [retorquere/zotero-better-bibtex#893]
My OS is Windows 7 64-bit
I am on Zotero Standalone 5.0.34, Better BibTex 5.0.78 and TexSudio 2.12.6
I have set Zotero preference-> Export->Default Format->Better BibTex Citation Key Quick Copy
# Bug classification
The drag and drop function does not work after starting Zotero.
The  **Edit->Copy as Better BibTex Citation Key Quick Copy**  or  **Hotkey Ctrl+Shift+C** work fine.
If I set the  **Zotero preference-> Export->Default Format** to other option and then set it back as **Better BibTex Citation Key Quick Copy**. the drag and drop will work. Nevertheless, if I close Zotero and start it again, the drag and drop function will be lost.

## Non-export problems with BBT
If your issue is a bug report, but not for exports, restart Zotero with debugging enabled (Help -> Debug Output Logging -> Restart with logging enabled), reproduce your problem, and select "Report Better BibTeX error" from the help menu, and post the resulting report ID (shown in red after you submit) here.

I cannot generate any bug report through the above steps.

**Comment by retorquere** [comment]
In what sense can you not generate a bug report? Is the option unavailable, do you get an error, something else?

**Comment by retorquere** [comment]
And after the fresh start, is the default format still set to `Better BibTex Citation Key Quick Copy`?

**Comment by rikeijin** [comment]
Dear Sir,
1. I follow the above steps and the report is empty, so when I push the Next button, I get a TypeError, cannot covert undefined to Object.
2. Yes, the default format is still `Better BibTex Citation Key Quick Copy`

**Comment by retorquere** [comment]
Can you go into `Help` -> `Debug Output Logging` -> `View Output`, copy-paste the content of the window that pops up into a text file, and attach that file here (do **not** paste the contents here -- attach it in a file by dragging the file on the issue here)

**Comment by retorquere** [comment]
Oh wait -- can you go into `Preferences` -> `Advanced` -> `Config Editor` -> `I accept the risk` -> search for `lockedinit` -> double-click `extensions.zotero.translators.better-bibtex.lockedInit` so it goes to `true`, and then restart and try again? If you do this, BBT will lock out Zotero until it is really ready. Maybe BBT isn't ready when you try the drag and drop.

**Comment by retorquere** [comment]
(but I'll also want a copy of that logfile)

**Comment by rikeijin** [comment]
Heare is the copy of logfile
[BBT_log.txt](https://github.com/retorquere/zotero-better-bibtex/files/1670376/BBT_log.txt)

**Comment by rikeijin** [comment]
I have tried` Preferences -> Advanced -> Config Editor -> I accept the risk -> search for lockedinit -> double-click extensions.zotero.translators.better-bibtex.lockedInit`
The drag and drop function still not work, so I copy another logfile as below after this try.

[BBT_log_2_after_locked.txt](https://github.com/retorquere/zotero-better-bibtex/files/1670382/BBT_log_2_after_locked.txt)

**Comment by retorquere** [comment]
OK, the error report should work again, but the other problem you have seems to be related to a [zotero problem](https://groups.google.com/forum/?pli=1#!topic/zotero-dev/j_FdSB3psmI) for which I haven't yet gotten any feedback. I will have to wait for feedback from the devs -- this doesn't seem to actually happen in BBT but in Zotero itself.

**Comment by retorquere** [comment]
Can you try doing an export of the reference using the `Better BibTex Citation Key Quick Copy` translator, and then doing a drag and drop?

**Comment by rikeijin** [comment]
The `Better BibTex Citation Key Quick Copy` translator work well (with one item or menay items), but after doing the export, drag and drop still does not work.
The report ID is **9MFN74W9**

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5457](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.0.79.5457.issue-893.xpi) ("preload export translators").

**Comment by retorquere** [comment]
Can you try 5457?

**Comment by rikeijin** [comment]
Dear Sir,
I have tried test build 5457. The problem is the same as decription initially: drag and drop does not work in a fresh start. However, if I change the `Zotero preference-> Export->Default Format` to other options such as `BibLaTex` and set it back to `Better BibTex Citation Key Quick Copy` , the drag and drop will work. The error report ID is **L78PTGFR**. Thank you for patience.

**Comment by retorquere** [comment]
I was afraid of that. We'll have to wait for the zotero devs to chime in - there's nothing bbt can do about this.

**Comment by mirgleich** [comment]
If that helps you.
To me it looks like I'm having the same Issue.
Already tried using lockedinit but that does not solve the issue. (Debian 9 with KDE GUI)
Report ID: ARJ2FP4J

**Comment by retorquere** [comment]
This really is a problem that needs assistance from the Zotero developers; I'm just supplying the translator, they are calling it, so there's not much I can do. I'm awaiting their feedback on [this thread](https://groups.google.com/forum/?pli=1#!topic/zotero-dev/j_FdSB3psmI).

**Comment by retorquere** [comment]
There have been new versions of Zotero out in the interim -- does this still occur with the latest Zotero?

**Comment by mirgleich** [comment]
no I'm stil getting an error in the comandline when trying drag and drop:
Zotero Version: Zotero Zotero 5.0.52, Copyright (c) 2006-2018 Contributors

```
Possibly unhandled rejection:

this._translatorInfo is null

TypeError: this._translatorInfo is null
    Zotero.Translate.Base.prototype._translateTranslatorLoaded<@chrome://zotero/content/xpcom/translation/translate.js:1365:30
From previous event:
    Zotero.Translate.Export.prototype.translate@chrome://zotero/content/xpcom/translation/translate.js:2571:10
    Zotero.BetterBibTeX<[219]/</<@chrome://zotero-better-bibtex/content/BetterBibTeX.js:551:12
    Zotero.QuickCopy</this.getContentFromItems@chrome://zotero/content/xpcom/quickCopy.js:261:4
    Zotero.ItemTreeView.prototype.onDragStart@chrome://zotero/content/xpcom/itemTreeView.js:2745:4
    ondragstart@chrome://zotero/content/standalone/standalone.xul:1:1
Possibly unhandled rejection:

Code promise is not resolved in noWait mode

Error: Code promise is not resolved in noWait mode
    Zotero.Translate.Base.prototype._loadTranslator<@chrome://zotero/content/xpcom/translation/translate.js:1807:12
    Zotero.Translate.Base.prototype.translate<@chrome://zotero/content/xpcom/translation/translate.js:1336:22
    Zotero.Translate.Export.prototype.translate@chrome://zotero/content/xpcom/translation/translate.js:2571:10
    Zotero.BetterBibTeX<[219]/</<@chrome://zotero-better-bibtex/content/BetterBibTeX.js:551:12
    Zotero.QuickCopy</this.getContentFromItems@chrome://zotero/content/xpcom/quickCopy.js:261:4
    Zotero.ItemTreeView.prototype.onDragStart@chrome://zotero/content/xpcom/itemTreeView.js:2745:4
    ondragstart@chrome://zotero/content/standalone/standalone.xul:1:1
```

**Comment by retorquere** [comment]
This will really require the zotero devs to weigh in because I can't do anything but drop in my translator, but my request [hasn't gotten any traction](https://groups.google.com/forum/#!searchin/zotero-dev/noWait$20mode%7Csort:date/zotero-dev/j_FdSB3psmI/fQd5vM-0BQAJ).

**Comment by rhuhle** [comment]
I can confirm this on Zotero 5.0.55 on Plasma 5, Gentoo linux:
>[JavaScript Error: "Translators not yet loaded"]
>[JavaScript Error: "Code promise is not resolved in noWait mode" {file: "chrome://zotero/content/xpcom/translation/translate.js" line: 1812}]
>
>[JavaScript Error: "this._translatorInfo is null" {file: "chrome://zotero/content/xpcom/translation/translate.js" line: 1370}]
>
>[JavaScript Error: "Code promise is not resolved in noWait mode" {file: "chrome://zotero/content/xpcom/translation/translate.js" line: 1812}]
>
>version => 5.0.55, platform => Linux x86_64, oscpu => Linux x86_64, locale => en-US, appName => Zotero, appVersion => 5.0.55, extensions => Zotero Better Bib(La)Tex (5.0.195, extension), Zotero LibreOffice Integration (5.0.14.SA.5.0.55, extension)
>
>(3)(+0000000): Scheduling next feed update
>
>(4)(+0000002): SELECT ( CASE WHEN lastCheck IS NULL THEN 0 ELSE strftime('%s', lastCheck) + refreshInterval * 60 - strftime('%s', 'now') END ) AS nextCheck FROM feeds WHERE refreshInterval IS NOT NULL ORDER BY nextCheck ASC LIMIT 1
>
>(3)(+0000004): No feeds with auto-update
>
>(4)(+0000002): SELECT libraryID AS id FROM feeds WHERE refreshInterval IS NOT NULL AND ( lastCheck IS NULL OR (julianday(lastCheck, 'utc') + (refreshInterval/1440.0) - julianday('now', 'utc')) <= 0 )
>
>(3)(+0000002): Running update for feeds:
>
>(3)(+0000002): All feed updates done
>
>(3)(+0000000): Scheduling next feed update
>
>(4)(+0000000): SELECT ( CASE WHEN lastCheck IS NULL THEN 0 ELSE strftime('%s', lastCheck) + refreshInterval * 60 - strftime('%s', 'now') END ) AS nextCheck FROM feeds WHERE refreshInterval IS NOT NULL ORDER BY nextCheck ASC LIMIT 1

**Comment by retorquere** [comment]
I know, but there's nothing I can do about it. It will require help from the zotero team, but I've not gotten feedback yet on my question on the zotero discussion group.

**Comment by retorquere** [comment]
(I prefer logs to be sent using the BBT "Send BBT Error Report" facility BTW)

**Comment by retorquere** [comment]
I think I have a fix for this in 5.1.8. You will need to restart one extra time after installation of 5.1.8 (or any version after that) but I *think* that will make the D&D problem go away. So 

* install 5.1.8, let it restart to complete installation
* Zotero starts picking up the old translators, BBT detects new translators must be installed and installs them
* Restart Zotero, which will now pick up the 5.1.8 translators; BBT will not install them again
